Predictions Pipeline: pak dependency resolver failing on scheduled runs since 2026-03-18

## Summary

The scheduled `Predictions Pipeline` cron (`0 8 * * 3`) has been failing on `main` every week since **2026-03-18**. Last green scheduled run was 2026-03-11.

## Root cause

Scheduled runs fail after ~18 seconds in the **Setup R dependencies** step with:

```
Error:
! error in pak subprocess
Caused by error:
  prop$stop_for_solution_error()
  pkgdepends:::pkgplan_stop_for_solve_error(self, private)
  base::throw(new_error("Could not solve package dependencies:\n", msg, ...))
```

Example failing run: [24127518493](https://github.com/peteowen1/panna/actions/runs/24127518493) (scheduled, 2026-04-08, 1m16s).

This is the `r-lib/actions/setup-r-dependencies@v2` step hitting an unsolvable dependency graph — likely a pinned transitive dep that conflicts after an upstream update.

## Secondary failure

The workflow_dispatch run on 2026-04-12 ([24295183054](https://github.com/peteowen1/panna/actions/runs/24295183054)) got past dependency setup (probably via cached packages) but failed at T+43m during **Export weekly PSR snapshots** with `##[error]The operation was canceled.` — either manually cancelled or hit the 75-min `timeout-minutes` boundary on a slow export. Worth confirming once the dep issue is unblocked.

## Suggested fix

1. Bump `cache-version` in the `setup-r-dependencies@v2` step to force a clean resolve.
2. If that still fails, inspect the `extra-packages:` list in `.github/workflows/predictions-pipeline.yml` — the `any::glmnet` / `any::xgboost` / `any::Matrix` trio is the most common source of pkgdepends conflicts.
3. Consider running `pak::pkg_deps_tree()` locally against the current `DESCRIPTION` + extra-packages list to see which edge the resolver is rejecting.

## Health context

Flagged red in `ops/reports/actions-health-2026-04-13.md` (30% success rate).

--- a/.github/workflows/predictions-pipeline.yml
+++ b/.github/workflows/predictions-pipeline.yml
@@ -44,7 +44,7 @@ jobs:
       - name: Setup R dependencies
         uses: r-lib/actions/setup-r-dependencies@v2
         with:
-          cache-version: 1
+          cache-version: 2
           extra-packages: |
             any::devtools
             any::piggyback
@@ -186,24 +186,6 @@ jobs:
             if (isTRUE(pipeline_failed)) quit(status = 1)
           "
 
-      - name: Export weekly PSR snapshots
-        continue-on-error: true
-        timeout-minutes: 50
-        env:
-          PANNADATA_DIR: ${{ runner.temp }}/pannadata
-          GITHUB_PAT: ${{ secrets.WORKFLOW_PAT }}
-        run: |
-          cat > /tmp/run_psr_weekly.R << 'EOF'
-          pannadata_path <- Sys.getenv("PANNADATA_DIR", "../pannadata/data")
-          opta_path <- file.path(pannadata_path, "opta")
-          devtools::load_all()
-          pannadata_dir(pannadata_path)
-          opta_data_dir(opta_path)
-          start_step <- "8b"
-          source("data-raw/estimated-skills/run_skills_pipeline.R")
-          EOF
-          Rscript /tmp/run_psr_weekly.R
-
       - name: Pipeline summary
         if: always()
         run: |
@@ -230,3 +212,21 @@ jobs:
           gh api repos/peteowen1/pannadata/dispatches \
             -f event_type=predictions-complete
           echo "Triggered build-blog-data.yml in pannadata"
+
+      - name: Export weekly PSR snapshots
+        continue-on-error: true
+        timeout-minutes: 50
+        env:
+          PANNADATA_DIR: ${{ runner.temp }}/pannadata
+          GITHUB_PAT: ${{ secrets.WORKFLOW_PAT }}
+        run: |
+          cat > /tmp/run_psr_weekly.R << 'EOF'
+          pannadata_path <- Sys.getenv("PANNADATA_DIR", "../pannadata/data")
+          opta_path <- file.path(pannadata_path, "opta")
+          devtools::load_all()
+          pannadata_dir(pannadata_path)
+          opta_data_dir(opta_path)
+          start_step <- "8b"
+          source("data-raw/estimated-skills/run_skills_pipeline.R")
+          EOF
+          Rscript /tmp/run_psr_weekly.R

